libminifi/src/core/ProcessSession.cpp

```diff
@@ -66,10 +66,12 @@ ProcessSession::~ProcessSession() {
 }
 
 void ProcessSession::add(const std::shared_ptr<core::FlowFile> &record) {
-  if (_updatedFlowFiles.find(record->getUUIDStr()) != _updatedFlowFiles.end()) {
+  utils::Identifier uuid;
+  record->getUUID(uuid);
+  if (_updatedFlowFiles.find(uuid) != _updatedFlowFiles.end()) {
```

Comment:
I would consider adding a version of `getUUID` that returns the result as a return value. It looks like `Identifier` can hold `Nil`, so that is something to keep in mind and something that makes returning even an `optional` unnecessary. I don't understand why the current version returns a bool of false in the return value instead of just returning a `Nil` UUID.

Reply:
the jury is still out on if we should consider `Nil` an `invalid` uuid, in the spec it is only mentioned as "special", but I am all for restricting our set of valid uuid's to not include `Nil` and then we can abolish all the `bool getUUID(Identifier&)` stuff

(the `getUUID` implementation itself already considers `Nil` to be invalid, so we just have to commit to this direction)

Reply:
if `Nil` is valid:
`getUUID` should return the `Nil` UUID, because it's a valid value, nothing special about them mandates special treatment. `getUUID` can just `return uuid_`.

if `Nil` is invalid:
`getUUID` can use `Nil` to represent an invalid value, instead of using `false`, simplifying the interface. Currently the only invalid value is `Nil`, so `getUUID` can just `return uuid_`.

Reply:
you are right, whatever we consider `Nil` to be (invalid or valid), the `getUUID` implementation is the same, created a ticket for refactoring our `Identifier` treatment e.g. don't pass `.getUUIDStr()` when a `.getUUID()` would be perfectly fine, and to replace all `.getUUID(uuid)` with the new implementation, I imagine there are quite a number of places that will be touched by that PR

the created ticket is [here](https://issues.apache.org/jira/browse/MINIFICPP-1395)